classes/class-woothemes-sensei-certificate-data-cleaner.php

```diff
@@ -0,0 +1,294 @@
+		'publish_certificate_templates',
+		'read_private_certificate_templates',
+		'delete_certificate_templates',
+		'delete_private_certificate_templates',
+		'delete_published_certificate_templates',
+		'delete_others_certificate_templates',
+		'edit_private_certificate_templates',
+		'edit_published_certificate_templates',
+	);
+
+	/**
+	 * Transient names (as MySQL regexes) to be deleted. The prefixes
+	 * "_transient_" and "_transient_timeout_" will be prepended.
+	 *
+	 * @var string[]
+	 */
+	private static $transients = array(
+		'sensei_certificates_job_create_certificates',
+	);
+
+	/**
+	 * User meta key names (as MySQL regexes) to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $user_meta_keys = array(
+		'^%BLOG_PREFIX%sensei_certificates_view_by_public$',
+	);
+
+	/**
+	 * Post meta to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $post_meta = array(
+		'_course_certificate_template',
+	);
+
+	/**
+	 * Cleanup all data.
+	 */
+	public static function cleanup_all() {
+		self::cleanup_custom_post_types();
+		self::cleanup_attachments();
+		self::cleanup_post_meta();
+		self::cleanup_capabilities();
+		self::cleanup_transients();
+		self::cleanup_settings();
+		self::cleanup_options();
+		self::cleanup_user_meta();
+
+		// Todo: Maybe cleanup the 'view-certificate' buttons.
+	}
+
+	/**
+	 * Checks if the uninstall option is enabled.
+	 *
+	 * @return bool
+	 */
+	public static function should_do_cleanup(): bool {
+		$settings = get_option( 'sensei-settings' );
+
+		return (bool) ( $settings['certificates_delete_data_on_uninstall'] ?? false );
+	}
+
+	/**
+	 * Cleanup data for custom post types.
+	 */
+	private static function cleanup_custom_post_types() {
+		foreach ( self::$custom_post_types as $post_type ) {
+			$items = get_posts(
+				array(
+					'post_type'   => $post_type,
+					'post_status' => 'any',
+					'numberposts' => -1,
+					'fields'      => 'ids',
+				)
+			);
+
+			foreach ( $items as $item ) {
+				wp_trash_post( $item );
+			}
+		}
+	}
+
+	/**
+	 * Cleanup data for attachments.
+	 */
+	private static function cleanup_attachments() {
+		global $wpdb;
+
+		foreach ( self::$attachment_guids as $attachment_guid ) {
+			$attachments = $wpdb->get_results(
+				$wpdb->prepare(
+					"SELECT ID FROM $wpdb->posts
+					WHERE post_type = 'attachment'
+					AND guid RLIKE %s",
+					$attachment_guid
+				)
+			);
+
+			foreach ( $attachments as $attachment ) {
+				wp_delete_attachment( $attachment->ID );
+			}
+		}
+	}
+
+	/**
+	 * Cleanup Sensei settings.
+	 */
+	private static function cleanup_settings() {
+		$settings = get_option( 'sensei-settings' );
+
+		if ( ! $settings ) {
+			return;
+		}
+
+		$settings_fields = WooThemes_Sensei_Certificates::instance()->certificates_settings_fields( [] );
+		foreach ( $settings_fields as $field_key => $field_data ) {
+			unset( $settings[ $field_key ] );
+		}
+
+		delete_option( 'sensei-settings' );
+		update_option( 'sensei-settings', $settings );
```

Comment:
For some reason, I had to delete the option before updating it. Otherwise, it's not updating. Not sure why...